### DISCUSSION TITLE
Format of ROM hex files?

### DISCUSSION THREAD

**Discussion by prlaba** [hneemann/Digital#945] [Q&A] [answered]
I’m a Logisim user, new to Digital, so forgive me for asking a very basic question.

What is the format of the file used to download the contents of a ROM component?  I’ve seen it referred to as a hex file but that doesn’t clarify the file’s actual format.  Is it Intel Hex format?  Some other hex format?  Does it require a special header line like Logisim?

It would help newbies like me if this information were included in Digital’s docs.  Maybe it’s there and I missed it.

Thanks!

**Comment by hneemann** [comment, marked as answer]
Both Logisim-HEX files and Intel-HEX files are supported.

**Reply by prlaba** [reply]
Thank you!

**Reply by prlaba** [reply]
Hi Helmut,

I tried downloading one of my Logisim hex files into a Digital ROM.  It doesn't appear that Digital supports Logisim's hex files, as it loaded each of the individual characters from the file (including Logisim's header line, spaces, etc.) as a single byte.  So the hex pair "00" in the file didn't get translated into the byte 0x00 and then loaded into the ROM, but instead got loaded as the two consecutive bytes 0x20 and 0x20.  White space and new line characters also got loaded "as is" as individual bytes.

So it seems Digital downloaded my Logisim hex file as a binary file, with each byte in the file mapped to a byte in the ROM.

Am I missing something?

Thanks,

**Comment by hneemann** [comment]
The file name must end with ".hex". Otherwise the file will be read as a plan binary file.

**Reply by prlaba** [reply]
Thanks for clarifying.

Is there any way for Digital to *not* be dependent on a specific file extension to differentiate Intel hex files, Logisim hex files and binary files?  Logisim, for example, doesn't require its hex files to have a .hex extension, so I generally use .lsm to indicate such a file.

Otherwise... Is this documented somewhere?  It would be good to include this sort of info in the Digital guide, to save you having to answer questions like mine.

**Comment by prlaba** [comment]
Uh, ok.

Is that documented anywhere?

Paul
